File: LeetCode/Easy/0014. Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
       int minLength = Integer.MAX_VALUE;
       int ind = 0;
       for (int i = 0; i < strs.length; i++) {
           if(strs[i].length() < minLength) {
               minLength = strs[i].length();
               ind = i;
           }
       }
       System.out.println(minLength);
       
       for(int i = 0; i < minLength; i++) {
           char c = strs[0].charAt(i);
           for(int j = 0; j < strs.length; j++) {
               if(strs[j].charAt(i) != c && i != 0) {
                   return strs[0].substring(0, i); 
               } else if(strs[j].charAt(i) != c) {
                   System.out.println("Here");
                   return "";
               }
           }
       }
       return strs[ind]; 
    }
}

File: LeetCode/Easy/0014. Longest Common Prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str

        idea:
         - find the common pref of strs[0] and strs[1] => common_pref
         - then find common pref of common_pref and strs[2] ...
        """

        if len(strs) == 1: return strs[0]

        common_pref = strs[0]

        for s in strs[1:]:
            while common_pref != s[:len(common_pref)]:
                common_pref = common_pref[:-1]
                if common_pref == "": return ""
        
        return common_pref
